### DISCUSSION TITLE
Regarding code capabilities

### DISCUSSION THREAD

**Discussion by Flyingsai** [xcompact3d/Incompact3d#230] [Q&A] [answered]
Respected Sir,

I am a beginner to Xcompact3d(that being said I have used incompact2d code given to me by my senior for 2 months), I would like to know about the capabilities of this newer code.  

**Background on my problem**: I am presently trying to run a DNS simulations matching an experimental investigation of a flow through adverse pressure gradient created by a contoured upper wall.

Instead of running a DNS simulation with contoured body placed in the upper boundary, I reasoned that by taking the experimental  velocity distribution created by the body(in the invisicd region), and applying it as a boundary condition, I could reduce the computational domain size.  

**My question**: Is there a provision to apply two different boundary conditions in the same directions(please see figure), i.e 

- 1.streamwise-velocity distribution boundary condition and no-slip boundary condition(both on x-z plane). 
- 2.F-S type velocity inlet profile and convective outlet (both on y-z plane) .

![diagram](https://github.com/xcompact3d/Incompact3d/assets/50273272/377bf567-8c00-486a-9c95-68d86d4a4e69)

Thank you for your time and consideration.

**Comment by slaizet** [comment]
Hi,
The answer is yes. You just need to use `ncly1=nclyn=2` (to allow for Dirichlet BC for the velocity) and then you just need to input the relevant BC subroutine (I guess in your case `subroutine boundary_conditions_tbl (ux,uy,uz,phi)`). I think you only need to change the top BC.
Best wishes,
Sylvain

**Reply by Flyingsai** [reply]
Thank u so much sir, I will try to implement it.

**Reply by Flyingsai** [reply]
Sir, does the code have the capability to create a free-slip contoured body (here on the upper domain), from my understanding incompact2d does not have this capability, please see the figure below.

![contoured body on upper domain](https://github.com/xcompact3d/Incompact3d/assets/50273272/78269fc1-cbb6-4a82-9a33-db4d0a8baaa1)

**Reply by slaizet** [reply, marked as answer]
Hi,
The computational domain needs to be a rectangle but it should be possible to impose a customised BC mimicking the set-up in the figure.
Best,
Sylvain
